src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdNewCaseCreator.java

```diff
@@ -39,33 +38,27 @@
     private final TransformationClient transformationClient;
     private final ServiceResponseParser serviceResponseParser;
     private final AuthTokenGenerator s2sTokenGenerator;
-    private final PaymentsProcessor paymentsProcessor;
     private final CoreCaseDataApi coreCaseDataApi;
-    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
 
     public CcdNewCaseCreator(
         TransformationClient transformationClient,
-        ServiceResponseParser serviceResponseParser, AuthTokenGenerator s2sTokenGenerator,
-        PaymentsProcessor paymentsProcessor,
-        CoreCaseDataApi coreCaseDataApi,
-        ExceptionRecordFinalizer exceptionRecordFinalizer
+        ServiceResponseParser serviceResponseParser,
+        AuthTokenGenerator s2sTokenGenerator,
+        CoreCaseDataApi coreCaseDataApi
     ) {
         this.transformationClient = transformationClient;
         this.serviceResponseParser = serviceResponseParser;
         this.s2sTokenGenerator = s2sTokenGenerator;
-        this.paymentsProcessor = paymentsProcessor;
         this.coreCaseDataApi = coreCaseDataApi;
-        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
     }
 
-    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
-    public ProcessResult createNewCase(
+    @SuppressWarnings("squid:S2139") // squid for exception handle + logging
+    public Either<ProcessResult, Long> createNewCase(
```

Comment:
This result type is a bit unclear. Maybe we can adjust ProcessResult to contain what we need? Or, if it's used in more places, we can have a version just for case creation.

Reply:
At the moment I don't see anything unclear here only missing docblock explaining it. i'll give it a go fresh tomorrow or will await for more input. and perhaps make assumptions in case there's a significant delay on _bugfix_

Reply:
`ProcessResult` class represents success or failure. Before the proposed changes, this method returned a result that could be returned straight to CCD. Now it returns either `ProcessResult` which only represents failure, or a bit of information that will be used for creating ProcessResult representing success. It's not very clear and needs a bit of analysis to figure out what's going on. How about replacing `ProcessResult` (just in `CcdNewCaseCreator`) with a different class, say `CaseCreationResult`, which would contain errors, warnings and new case ID, and make the calling code always prepare `ProcessResult` out of it?

I like the fact that you moved the use of `ExceptionRecordFinalizer` to the calling method - now we consistently prepare exception record data there.

Reply:
appreciate such response. will proceed with changes sooner than later ;)